chat_stream.go

```diff
@@ -0,0 +1,106 @@
+package gogpt
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"encoding/json"
+	"io"
+	"net/http"
+)
+
+type ChatCompletionStreamResponse struct {
+	ID      string `json:"id"`
+	Object  string `json:"object"`
+	Created int64  `json:"created"`
+	Model   string `json:"model"`
+	Choices []struct {
```

Comment:
Can we avoid inline structs here, please?

Reply:
No problem.

Reply:
> Can we avoid inline structs here, please?

Completed.

Reply:
Thanks!